avocado/core/magic.py

```diff
@@ -0,0 +1,17 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; specifically version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat, Inc. 2020
+# Author: Cleber Rosa <crosa@redhat.com>
+
+
+#: Magic number used to check that an "avocado" command is indeed
+#: present and matches one's expectations
+MAGIC = "971dfda4234b6abbea36ed1e939664f45c0190b234918d627449c8b578f8072e"
```

Comment:
Hmmm, I was wondering the same. And usage of MAGIC of any kind is generally better avoided in software patterns.

Reply:
The idea is we want to make sure that "our" Avocado is present.

For comparison purposes, our older `runner-{remote,vm,docker}` plugins, would run `avocado --version` and check if `Avocado \d\d\.\d` (or something like this) was present.  But there could be a different application with the same name installed (on PyPI alone we have quite a lot of similar named packages: https://pypi.org/project/avocado/, https://pypi.org/project/avocado-messenger-server/).

So, instead of doing more heuristics, such as trying to find the right packages or files, I thought a magic number would avoid any confusion.

Reply:
I think looking for a binary or an importable module of a particular version the way it was done before is a much more standard approach to check if a given dependency is provided. In this case `avocado --version` should be good enough for most practical situations. Are there important use cases where we need to verify greater authenticity for the avocado code? I guess in such extreme cases the magic hex number will not work either.

Reply:
Yes, I understand but there is a lot of code here just to print a MAGIC number (plus a different subcommand).

So, my suggestion is to print a small hash along with `avocado --version` and keep things simples.

Reply:
Btw, how is a small hash in addition to the version contributing more, i.e. how is the version by itself not unique enough to find the avocado dependency and a versioned one in particular?

Reply:
@pevogam Yes, there are no advantages (unless I'm missing something) with the simple version. We could check if `version >= 80` for instance. So, I'm ok with a simple version too.

Reply:
OK, so let's drop this, and if the need arises, we can *add* a more specific "fingerprint" print/check.